### ISSUE TITLE
AWSQS::Kubernetes::Resource can't recreate the kubernetes-resource-proxy after deleting an addon

### ISSUE THREAD

**Issue by tomiszili** [aws-quickstart/quickstart-kubernetes-resource-provider#33]
Hello!

The AWSQS::Kubernetes::Resource deletes the kubernetes-resource-proxy after deleting a manifest with them.
In my case i deployed the metrics server addons and delete the addon for test purposes, but when I want to deploy it again the CloudFormation fails. I paste the logs of the `awsqs-kubernetes-resource` and it can't find the `kubernetes-resource-proxy-<clustername>` lambda because it was deleted from the previous CloudFormation addon delete. And the next time it fails to recreate the function.

I think some modification makes the zip files too large for the lambda_create function `ZipFile` parameter.

```
Exception caught An error occurred (RequestEntityTooLargeException) when calling the CreateFunction operation: Request must be smaller than 69905067 bytes for the CreateFunction operation
Traceback (most recent call last):
  File "/var/task/awsqs_kubernetes_resource/vpc.py", line 116, in put_function
    no_update = update_function_config(lmbd, function_config)
  File "/var/task/awsqs_kubernetes_resource/vpc.py", line 131, in update_function_config
    lmbd.update_function_configuration(**function_config)
  File "/var/task/botocore/client.py", line 514, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/var/task/botocore/client.py", line 938, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.ResourceNotFoundException: An error occurred (ResourceNotFoundException) when calling the UpdateFunctionConfiguration operation: Function not found: arn:aws:lambda:eu-central-1:201397412937:function:awsqs-kubernetes-resource-proxy-Demo-1v21

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/task/cloudformation_cli_python_lib/resource.py", line 221, in __call__
    raise error
  File "/var/task/cloudformation_cli_python_lib/resource.py", line 214, in __call__
    progress = self._invoke_handler(caller_sess, request, action, callback)
  File "/var/task/cloudformation_cli_python_lib/resource.py", line 94, in _invoke_handler
    progress = handler(session, request, callback_context)
  File "/var/task/awsqs_kubernetes_resource/handlers.py", line 74, in create_handler
    session,
  File "/var/task/awsqs_kubernetes_resource/utils.py", line 53, in run_command
    put_function(session, cluster_name)
  File "/var/task/awsqs_kubernetes_resource/vpc.py", line 121, in put_function
    lmbd.create_function(Code={"ZipFile": zip_file.read()}, **function_config)
  File "/var/task/botocore/client.py", line 514, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/var/task/botocore/client.py", line 938, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (RequestEntityTooLargeException) when calling the CreateFunction operation: Request must be smaller than 69905067 bytes for the CreateFunction operation
```
This behaviour affects [this issue](https://github.com/aws-quickstart/quickstart-amazon-eks/issues/452) as well.

**Comment by gargana** [comment]
Working on a fix for this.

**Comment by tomiszili** [comment]
After I added into the requirements.txt the following `cryptography==3.4.8` the error mentioned above seems fixed.

But now I got new error messages after I want to install the metrics-server.
```
Code: FileNotFoundError Message: [Errno 2] No such file or directory: 'aws': 'aws'
```

```
StackTrace: ['  File "/var/task/awsqs_kubernetes_resource/utils.py", line 206, in proxy_wrap\n    create_kubeconfig(event["cluster_name"])\n', '  File "/var/task/awsqs_kubernetes_resource/utils.py", line 97, in create_kubeconfig\n    None,\n', '  File "/var/task/awsqs_kubernetes_resource/utils.py", line 67, in run_command\n    shlex.split(command), stderr=subprocess.STDOUT\n', '  File "/var/lang/lib/python3.7/subprocess.py", line 411, in check_output\n    **kwargs).stdout\n', '  File "/var/lang/lib/python3.7/subprocess.py", line 488, in run\n    with Popen(*popenargs, **kwargs) as process:\n', '  File "/var/lang/lib/python3.7/subprocess.py", line 800, in __init__\n    restore_signals, start_new_session)\n', '  File "/var/lang/lib/python3.7/subprocess.py", line 1551, in _execute_child\n    raise child_exception_type(errno_num, err_msg, err_filename)\n']
```

```
Exception caught FileNotFoundError: [Errno 2] No such file or directory: 'aws': 'aws'
Traceback (most recent call last):
  File "/var/task/cloudformation_cli_python_lib/resource.py", line 219, in __call__
    raise error
  File "/var/task/cloudformation_cli_python_lib/resource.py", line 212, in __call__
    progress = self._invoke_handler(caller_sess, request, action, callback)
  File "/var/task/cloudformation_cli_python_lib/resource.py", line 92, in _invoke_handler
    progress = handler(session, request, callback_context)
  File "/var/task/awsqs_kubernetes_resource/handlers.py", line 148, in delete_handler
    if not get_model(model, session):
  File "/var/task/awsqs_kubernetes_resource/utils.py", line 222, in get_model
    outp = run_command(cmd, cluster, session)
  File "/var/task/awsqs_kubernetes_resource/utils.py", line 58, in run_command
    resp = proxy_call(cluster_name, manifest, command, session)
  File "/var/task/awsqs_kubernetes_resource/vpc.py", line 69, in proxy_call
    raise Exception(f'{resp["errorType"]}: {resp["errorMessage"]}')
Exception: FileNotFoundError: [Errno 2] No such file or directory: 'aws': 'aws'
```

Any suggestion to fix this?

**Comment by gargana** [comment]
Just looking at the error detail provided I see `/var/lang/lib/python3.7/subprocess.py`. 

The initial function the Quick Start creates should be created as `python 3.8`.
 https://github.com/aws-quickstart/quickstart-amazon-eks/blob/main/templates/amazon-eks-functions.template.yaml#L97-L118
 
 Can you confirm that this is happening only with `EKS 1.23`? 
 
 Also it isn't clear to me where you updated `cryptography==3.4.8`? Could you please clarify?
 
 Thanks for the detailed info.

**Comment by tomiszili** [comment]
I have deleted all the previous quickstart templates and resources from my account, as well as deleted all the Cloudformation privately registered extensions, and then I tried the whole EKS quickstart deployment again with the latest KubernetesResource templates.
Now its works properly with 1.23 as well, it deploys the KubernetesResrouce function with python3.8, and know i didn't got any errors even with the initial FuncktionStack's version of the lambda. 

I don't know why it created eralier the function with python3.7... I have started to debug the errors i got so that's why i downgraded the cryptography to a fix version, but is wasn't the solution, so that was a wrong idea.

Sorry for the misleading and unclear comments.
